src/routes/+layout.svelte

```diff
@@ -42,6 +47,9 @@
 	}
 
 	onMount(() => {
+		// must be running in the browser to interact with the document
+		new DateTriggerEventsHandler(DEFAULT_EVENTS);
```

Comment:
won't this cause style flashes while loading?

Reply:
yea, there's no other way to do it, at least not of my knowledge

Reply:
put a <script> tag in the html - it's standard practice if you must have custom theming

Reply:
osumatrix told me to try and use npm instead of a script tag, but sure, if that's the only way, I'm down

Reply:
image shows the request to the url is successful, but the file is empty, even tho I can reach it and it clearly has contents
![image](https://github.com/ReVanced/revanced-website/assets/100418457/56797385-4afd-430f-893d-12d14d96a4e7)

Reply:
using unpkg works, idk why gh doesn't. is this what you mean?
![image](https://github.com/ReVanced/revanced-website/assets/100418457/29c3aea1-65a3-43d5-9838-18e0b5107f91)

Reply:
I have explained why it is a better idea to use NPM.

It is not a good idea to load an external script through a <script>. For one it makes a outgoing request, for two, you don't pin the version and although possible, it would be better to bundle it with the current build for speed.

Reply:
ok but is there any way to have the same speed as a script tag? because even tho onMount is fast, it's not _as_ fast as a script, which is almost instantaneous

Reply:
I am not seeing how loading an external file, unoptimized for the current project at compile time and more, is faster. The remaining reasons also argue against script.

Reply:
I have no clue why,bill test again later on a different network, but the speed difference is very noticeable, the CSS changes are almost instantaneouswith the script tag, but with onMount they're delayed almost a whole second

Reply:
The technicality with onMount is not at question here. Its about network speed.

Reply:
@osumatrix I think you misunderstood a bit.

I'm not saying to load any libraries from the script tag. I'm saying that the script should be inlined in the HTML so there's no network delay. As stated before this is *the* way to prevent theme flashes.

Reply:
@KTibow 
> I'm saying that the script should be inlined in the HTML so there's no network delay. As stated before this is _the_ way to prevent theme flashes.

if I put the script tag in the head and leave the imports in +layout.svelte then it won't work

![image](https://github.com/ReVanced/revanced-website/assets/100418457/7c0268d2-9789-48e1-b1af-e4d0898dac07)
![image](https://github.com/ReVanced/revanced-website/assets/100418457/722ed36b-b0eb-47bc-88f6-748f312d98ac)

Reply:
Yup, imports aren't shared globally. You would need to inline it here.

Reply:
@KTibow doesn't work anyways :P
![image](https://github.com/ReVanced/revanced-website/assets/100418457/42323586-12db-4e90-b82d-00a65175ca81)
![image](https://github.com/ReVanced/revanced-website/assets/100418457/7751d8e5-168c-431d-bfc8-0c1cc0e700f0)

Reply:
As I stated, the imports aren't shared. The import won't magically work its way into the script tag, as you might be used to. Each script tag is treated as its own module thanks to bundling.

Reply:
@KTibow I know but what I don't know is how to "inline" it like you told me to

Reply:
@KTibow I know but what I don't know is how to "inline" it like you told me to

Reply:
Re: bundle size

While this library is great and has a logical structure, it would probably need to be simplified to reduce its size.
4.8kb minified / 2.2kb gzip'd is a bit large for something that could be represented as
```js
(() => {
const colors = {
  new_year: {text: "#e6e8fe", background: "#010215", primary: "#7082f9", secondary: "#960670", accent: "#f62e65"},
  valentines_day: {text: "#efd6f9", background: "#18051f", primary: "#dc8fef", secondary: "#93166c", accent: "#e34190"},
  april_fools: {text: "#ceeafe", background: "#000910", primary: "#7ec2fc", secondary: "#9504af", accent: "#fa28a7"},
  halloween: {text: "#f2eeed", background: "#0c0705", primary: "#df9d87", secondary: "#923112", accent: "#fd490e"},
  christmas: {text: "#fdfcfc", background: "#030202", primary: "#0ee768", secondary: "#f30404", accent: "#ffffff"},
  easter: {text: "#f7defc", background: "#09010c", primary: "#dc7df2", secondary: "#f60087", accent: "#6cdeb8"},
}
function getEaster(Y) {
  const a = Y % 19,
      b = Math.floor(Y / 100),
      c = Y % 100,
      d = Math.floor(b / 4),
      e = b % 4,
      f = Math.floor((b + 8) / 25),
      g = Math.floor((b - f + 1) / 3),
      h = (19 * a + b - d - g + 15) % 30,
      i = Math.floor(c / 4),
      k = c % 4,
      L = (32 + 2 * e + 2 * i - h - k) % 7,
      m = Math.floor((a + 11 * h + 22 * L) / 451),
      month = Math.floor((h + L - 7 * m + 114) / 31),
      day = ((h + L - 7 * m + 114) % 31) + 1;
  return new Date(Y, month - 1, day);
}
function getEvent() {
  const date = new Date();
  const easter = getEaster(date.getFullYear());
  const month = date.getMonth() + 1;
  const day = date.getDay();
  if (month == 1 && day == 1) return "new_year";
  if (month == 2 && day == 14) return "valentines_day";
  if (month == 4 && day == 1) return "april_fools";
  if (month == 10 && day == 31) return "halloween";
  if (month == 12 && day == 25) return "christmas";
  if (date.getMonth() == easter.getMonth() && date.getDate() == easter.getDate()) return "easter";
}
function setColors(colors) {
  document.body.style.setProperty("--text", colors.text);
  document.body.style.setProperty("--background", colors.background);
  document.body.style.setProperty("--primary", colors.primary);
  document.body.style.setProperty("--secondary", colors.primary);
  document.body.style.setProperty("--accent", colors.accent);
}
function update() {
  const event = getEvent();
  const c = colors[event];
  if (c) setColors(c);
  else setColors({text: null, background: null, primary: null, secondary: null, accent: null});
}
update();
setInterval(update, 1000 * 60);
})()
```
That's 1.6kb minified, or 718 bytes gzip'd. I could probably code golf it more if I wanted but I'm satisfied with this minimal implementation.

Reply:
so sould I just rewrite the whole lib to be functional instead of OOP?

Reply:
your original lib is nice and customizable, you shouldn't change that.
however, when we're only trying to do one thing we can take shortcuts like making it less extensible, which make it take up less size (see above)

Reply:
sure, I'll take care of that

Reply:
@KTibow is your intention to move the code into the website? The less on our shoulders the better. Artificial speed differences are second priority

Reply:
just tell me what's y'all choice and I'll implement it. I've already simplified the Easter function thanks to you and removed an entire file, which made the lib way smaller

Reply:
> The size we are talking about is less than a single image file.

Images can be lazy loaded. This needs to run *instantly*, as soon as the website loads.

> if we remove and tightly integrate into the website would stop to exist and intertwine the current codebase more than it should

I'm confused by this English. What's stopping to exist? What's "it" here?

> which made the lib way smaller

Upload to NPM so we can see it then

Reply:
@KTibow I still don't get how to "inline" the code, can you explain it to me?